modules/Auth/lib/ReservationHandler.cpp

```diff
@@ -253,19 +248,14 @@ ReservationHandler::cancel_reservation(const int reservation_id, const bool exec
 
     std::pair<bool, std::optional<uint32_t>> result;
 
-    std::unique_lock<std::recursive_mutex> lock(reservation_mutex);
```

Comment:
This function is also called from the timer within the reservation handler (see line 581, in 'set_reservation_timer'). So I think the mutex should not be removed here.

Reply:
I added that one again